core/chaincode/lifecycle/deployedcc_infoprovider_test.go

```diff
@@ -236,6 +240,114 @@ var _ = Describe("ValidatorCommitter", func() {
 		})
 	})
 
+	Describe("AllChaincodesInfo", func() {
+		var (
+			fakeStateIteratorKVs MapLedgerShim
+		)
+		BeforeEach(func() {
+			fakeStateIteratorKVs = MapLedgerShim(map[string][]byte{})
+			err := resources.Serializer.Serialize("namespaces", "cc-name", &lifecycle.ChaincodeDefinition{}, fakeStateIteratorKVs)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Better to add a non-nil collection config in `lifecycle.ChaincodeDefinition`, because the function `AllChaincodesInfo` is expected to return that as well.

Reply:
The empty ChaincodeDefinition here is only used for GetStateRangeScanIteratorStub so that QueryNamespaceDefinitions can return a mocked map[string]string. The actual ChaincodeDefinition is from mocked GetState that returns fakeChaincodeDef (line 94). In the test below BeforeEach, it verifies the result against fakeChaincodeDef.